File: Lab2/lab2/src/main/java/org/example/Result.java
package org.example;

public class Result {
    private float[][] matrix;
    public Result(int matrixSize)
    {
        matrix = new float[matrixSize][matrixSize];
    }

    public Result(float[][] matrix)
    {
        this.matrix = matrix;
    }

    public float[][] getMatrix() {
        return matrix;
    }

    public void WriteValueToCell(float cellValue, int row, int column)
    {
        matrix[row][column] = cellValue;
    }

    public void AddValueToCell(float cellValue, int row, int column)
    {
        matrix[row][column] += cellValue;
    }

    public void AddSubMatrix(float[][] subMatrix, int row, int column)
    {
        for (int i = 0; i < subMatrix.length; i++)
        {
            for (int j = 0; j < subMatrix[0].length; j++)
            {
                matrix[row + i][column + j] += subMatrix[i][j];
            }
        }
    }
}

File: Lab8/MatrixMultiplicationWebApp/MatrixMultiplicationWebApp.Core/Result.cs
namespace MatrixMultiplicationWebApp.Core;

public class Result {
    private readonly int[][] _matrix;

    public Result(int matrixSize)
    {
        _matrix = new int[matrixSize][];
        for (int i = 0; i < _matrix.Length; i++)
        {
            _matrix[i] = new int[matrixSize];
        }
    }

    public int[][] GetMatrix() {
        return _matrix;
    }

    public void WriteValueToCell(int cellValue, int row, int column)
    {
        _matrix[row][column] = cellValue;
    }
}